Share StaticDataSourceConfiguration between tests

orangejenny opened the pull request:
https://github.com/dimagi/commcare-hq/pull/23874/files#diff-922e9297b3c8138d87abebf772e70923R64

Pulling all `StaticDataSourceConfiguration` is surprisingly (to me) slow; it takes a couple of seconds. This shares it among all of the tests in each class, which only ever read from it.

orangejenny committed:
Combined usage of StaticDataSourceConfiguration.all() in TestStaticDataSource

orangejenny committed:
Combined usage of StaticDataSourceConfiguration.all() in TestStaticReportConfig

emord reviewed (approved):
> Pulling all StaticDataSourceConfiguration is surprisingly (to me) slow;

Agreed. Its easy to unintentionally add to this time by copying a new domain into an existing data source.

(Mostly just saying this to share it publicly) My goal is to remove domains and server environments from the static data source configurations and introduce a table to handle domain mappings on an environment. That could also fix issues where adding a new domain takes a code change/deploy. I've been hoping to PID that for a while, but haven't yet made the time

orangejenny committed:
Fixed test_static_reports

Don't use cls.configs in places that are testing test sets of configs,
but do use it in test for data source existence.

orangejenny committed:
Fixed test_static_data_sources

Let the tests that use test configurations use those configurations.

orangejenny committed:
Fixed lint error

orangejenny committed:
Switched patch path
